### PR TITLE
Version 4.1 Release Notes

### PR DESCRIPTION
**Pull request by mtopolnik**
Add description here

Checklist
- [x] Tags Set
- [x] Milestone Set

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/82/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/82/jet-oss-pr-builder-docs-no-build-82.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/83/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/83/jet-oss-pr-builder-docs-no-build-83.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Comment by mtopolnik** [comment]
verify

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 1]
```diff
@@ -1,9 +1,74 @@
...
+@caioguedes
+@SapnaDerajeRadhakrishna
+
+Thank you for your valuable contributions!
+
+1. New Features
+
+[jms] Exactly-once guarantee for JMS sink (#1813)
+[jdbc] Exactly-once guarantee for JDBC sink (#1813)
+[core] JobConfig.addClass() automatically adds nested classes to the job (#1932, #2077)
```
Comment (cangencer) [inline hazelcast-jet-distribution/src/root/release_notes.txt:41]: what about addPackage?

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 2]
```diff
@@ -1,9 +1,74 @@
...
+    fat JAR in the distribution archive (#2079)
+[pipeline-api] First-class support for inner hash join (@TomaszGaweda #2089)
+[core] When Jet starts up, it now logs the cluster name (@caioguedes #2105)
+[core] Add useful error message when trying to deploy a JDK class with JobConfig (#2108)
+[core] Implement JobConfig.toString (@SapnaDerajeRadhakrishna #2152)
+[core] Do not destroy Observable on shutdown (#2170)
+
+3. Fixes
+
+[core] Avoid thread interruption when terminating (#1971)
```
Comment (cangencer) [inline hazelcast-jet-distribution/src/root/release_notes.txt:61]: when terminating what?

**Comment by cangencer** [comment]
what about changelog.txt?

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 3]
```diff
@@ -1,9 +1,75 @@
...
-in Hazelcast Jet GitHub repository. You can reach to the full descriptions of the
-issues at https://github.com/hazelcast/hazelcast-jet/issues/<issue number>.
 
-The label in the brackets refer to the respective module.
+
+                    Hazelcast Jet Release Notes
+
+This document keeps the history of Hazelcast Jet releases. We try to
+enumerate all the non-trivial changes done to the product. The changes
+are catogerized into:
```
Comment (Serdaro) [inline hazelcast-jet-distribution/src/root/release_notes.txt:7]: .. into the following:

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 4]
```diff
@@ -1,9 +1,75 @@
...
+
+    1. New features
+    2. Enhancements to the existing features
+    3. Fixes of bugs and other problems
+    4. Breaking changes: those that result either in compile errors or changes
```
Comment (Serdaro) [inline hazelcast-jet-distribution/src/root/release_notes.txt:12]: 4. Breaking changes that result in ...

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 5]
```diff
@@ -1,9 +1,75 @@
...
+       in runtime behavior
+
+The items in the lists follow this format:
+
+[module] Description [PR_or_issue_number]
+
+For major features that have an associated design focument, [TDD_number]
```
Comment (Serdaro) [inline hazelcast-jet-distribution/src/root/release_notes.txt:19]: ... design document ...

File: hazelcast-jet-distribution/src/root/release_notes.txt (hunk 4 above)
Comment (mtopolnik) [reply hazelcast-jet-distribution/src/root/release_notes.txt:12]: That would change the meaning, though. I can put it it parentheses.

**Pull request closed without merging**